src/basics.md

```diff
@@ -470,6 +471,80 @@ fn main() {
 }
 ```
 
+[ex-phone]: #ex-phone
+<a name="ex-phone"></a>
+## Extract phone numbers from text
+
+Attempts to process a static string as a list including US phone numbers.  The
+expression is checking that there are 7 digits that come in groups of 3 3 4,
+separated by spaces, dots, dashes, and parenthesis.  The phone number is then
+tested based on formatting rules.
+
+```rust
+# #[macro_use]
+# extern crate error_chain;
+extern crate regex;
+
+use regex::Regex;
+use std::fmt;
+
+# error_chain!{
+#     foreign_links {
+#         Regex(regex::Error);
+#         Io(std::io::Error);
+#     }
+# }
+#
+struct PhoneNumber {
+    country: String,
+    number: String,
+}
+
+// This allows printing phone numbers based on country convention.
+impl fmt::Display for PhoneNumber {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self.country.as_ref() {
+            "US" => write!(f, "+1 ({}) {}-{}", &self.number[0..3], &self.number[3..6], &self.number[6..10]),
+            _    => write!(f, "{}", self.number),
+        }
+    }
+}
+
+fn run() -> Result<()> {
+    let phone_text = "
+    +1 505 881 9292
+    (202) 991 9534
+    Alex 5553920011
+    1 (800) 233-2010
+    1.299.339.1020";
+    let re = Regex::new(r"(?x)
+                         \+?1?                      #Country Code Optional
+                         [\x20\(\.]*
```

Comment:
- yeah I would go with `\s`
- `\+?1? `  should be moved int an optional group `(\+1)?\s*` so that plus char has to be followed by 1otherwise these two are independent
- I meant that the area codes expression within the parenthesis has to be matched with both opening and closing parens as a whole or | without both of them. currently the match for opening ,closing parens are independent so we can match `(((202 991 9534` .

 I may try to help but writing regexes on mobile is a real pain and I have no way to give you the full answer or even verify if my suggestions are fully sane. So you might try to solving this yourself or wait for me to get back behind a keyboard within next two weeks 😢 .